### PR TITLE
querier, store-gateway: promote chunks streaming to stable

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
#### What this PR does

preparing for the 2.14 release

#### Which issue(s) this PR fixes or relates to

related to https://github.com/grafana/mimir/issues/8390

#### Checklist

- [x] Tests updated.
- [x] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [x] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by pracucci — APPROVED** [review]

**Review by tacole02 — APPROVED** [review]

File: integration/querier_test.go [hunk 1]
```diff
@@ -234,11 +225,8 @@ func testQuerierWithBlocksStorageRunningInMicroservicesMode(t *testing.T, stream
 			// Make sure the querier is using the bucket index blocks finder.
 			require.NoError(t, querier.WaitSumMetrics(e2e.Greater(0), "cortex_bucket_index_loads_total"))
 
-			comparingFunction := e2e.Equals
-			if streamingEnabled {
-				// Some metrics can be higher when streaming is enabled. The exact number is not deterministic in every case.
-				comparingFunction = e2e.GreaterOrEqual
-			}
+			// Some metrics can be higher when streaming is enabled. The exact number is not deterministic in every case.
+			comparingFunction := e2e.GreaterOrEqual
```
Comment (charleskorn) [inline integration/querier_test.go:229]: This should be deterministic as of https://github.com/grafana/mimir/pull/8039

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@
 * [CHANGE] Store-gateway: enabled `-blocks-storage.bucket-store.max-concurrent-queue-timeout` by default with a timeout of 5 seconds. #8496
 * [CHANGE] Store-gateway: enabled `-blocks-storage.bucket-store.index-header.lazy-loading-concurrency-queue-timeout` by default with a timeout of 5 seconds . #8667
 * [CHANGE] Distributor: Incoming OTLP requests were previously size-limited by using limit from `-distributor.max-recv-msg-size` option. We have added option `-distributor.max-otlp-request-size` for limiting OTLP requests, with default value of 100 MiB. #8574
+* [CHANGE] Querier, store-gateway: remove deprecated `-querier.prefer-streaming-chunks-from-store-gateways=true`. #8696
```
Comment (56quarters) [inline CHANGELOG.md:13]:
```suggestion
* [CHANGE] Querier, store-gateway: remove deprecated `-querier.prefer-streaming-chunks-from-store-gateways=true`. Streaming from store-gateways is now always enabled. #8696
```

**Review by 56quarters — APPROVED** [review]

File: integration/querier_test.go (hunk 1 above)
Comment (dimitarvdimitrov) [reply integration/querier_test.go:229]: this was me deleting code without reading it. Thanks for pointing it out

Comment (dimitarvdimitrov) [reply integration/querier_test.go]: thanks for pointing this out. I replaced it with `Equals`, than the test locally 4 times and there were no flakes.

**Review by charleskorn — APPROVED** [review]

**Pull request merged**